add method allowing for multiple rows in connection::values

kouta-kun opened the pull request:
The current DuckDBPyConnection::Values method doesn't allow for multiple rows (passing something like `[[1,2],[3,4]]` generates a single row with 2 columns instead of 2 rows with 2 columns each). I've added a value_rows method which takes in a list of lists in which each one is a new row.

kouta-kun committed:
add method allowing for multiple rows in connection::values

Mause commented on tools/pythonpkg/src/include/duckdb_python/pyconnection.hpp:
```diff
@@ -74,7 +74,7 @@ struct DuckDBPyConnection {
 
 	unique_ptr<DuckDBPyRelation> Values(py::object params = py::none());
-
+        unique_ptr<DuckDBPyRelation> ValuesRows(py::list params = py::none());
```
Fix the formatting here please - should be enough to run `make format-fix`

Mause replied:
Also, why is `params` optional?

Mause commented on tools/pythonpkg/src/pyconnection.cpp:
```diff
@@ -75,6 +75,8 @@ void DuckDBPyConnection::Initialize(py::handle &m) {
 	    .def("values", &DuckDBPyConnection::Values, "Create a relation object from the passed values",
 	         py::arg("values"))
+	    .def("value_rows", &DuckDBPyConnection::ValuesRows, "Create a relation object from the passed values (as a rowset)",
+	         py::arg("values"))
```
This argument would probably be better named `rows`?

kouta-kun replied:
I've updated it to such in the latest commit

Mause commented on tools/pythonpkg/src/pyconnection.cpp:
```diff
@@ -299,6 +301,23 @@ unique_ptr<DuckDBPyRelation> DuckDBPyConnection::Values(py::object params) {
+	if (params.is_none()) {
+		params = py::list();
+	}
+	if (!py::hasattr(params, "__len__")) {
```
This isn't enough to determine if something is `iterable` (`__getitem__` or `__iter__` is also required) - you can probably do `py::isinstance<py::iterable>(...)` instead

kouta-kun replied:
It's optional only because I've copied over from Values' definition. I'm guessing the idea was to get an empty row but I wouldn't know what that was for.

I've replaced the checks with py::isinstance and I've also applied it to each of the items of the iterable (value_rows shouldn't be used in the same manner as values is, so passing in a list of values will fail)

Mause reviewed (changes requested):
Can we please see some tests added as well?

Mause commented:
Oh, you'll also need to update the type stubs (pyi file) as well, `./scripts/regenerate_python_stubs.sh` should do it

kouta-kun committed:
format fix

kouta-kun committed:
add tests, update stubs

kouta-kun commented:
I've made the requested changes and updated the type stubs but I get
```
Stubs must be updated, either add them to skip_stubs_errors or update __init__.pyi accordingly
[': not checking stubs due to mypy build ']
```
I'm guessing the github actions build won't get this error?

Also I've added a single test, verifying that value_rows does generate two rows from [[1,2],[3,4]] while values generates a single one from the same values.

kouta-kun commented:
Hi! @Mause are there any blockers following the last commits?

Mytherin commented:
The main blocker here seems to be the broken CI/merge conflicts. Are you interested in fixing those/picking it up again?

Mytherin commented:
Closing for now
